pandas/core/groupby/indexing.py

```diff
@@ -0,0 +1,289 @@
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING,
+    Iterable,
+    cast,
+)
+
+import numpy as np
+
+from pandas._typing import PositionalIndexer
+from pandas.util._decorators import (
+    cache_readonly,
+    doc,
+)
+
+from pandas.core.dtypes.common import (
+    is_integer,
+    is_list_like,
+)
+
+if TYPE_CHECKING:
+    from pandas import (
+        DataFrame,
+        Series,
+    )
+    from pandas.core.groupby import groupby
+
+
+class GroupByIndexingMixin:
+    """
+    Mixin for adding ._body to GroupBy.
+    """
+
+    @cache_readonly
+    def _body(self) -> BodyGroupByIndexer:
```

Comment:
can we rename to something else, maybe _positional_selector (e.g. just something more descriptive)